code/modules/shuttle/shuttle.dm

```diff
@@ -226,6 +248,13 @@
 
 	return 1
 
+/obj/docking_port/mobile/Destroy()
+	if(i_know_what_im_doing)
+		shuttle_master.mobile -= src
+		areaInstance = null
+		destination = null
+		previous = null
+	. = ..()
```

Comment:
If @Fox-McCloud is bugging kyep about this needing to be `return ..()`, I guess I have to do the same here